main.js

```diff
@@ -15,41 +15,58 @@ function enterKeyHandler() {
 	//console.log(event);
 	if ( event.key === "Enter") {
 		//console.log("you hit enter");
-		let optObj = createOptionsObject();
-		//console.log("here's your options object", optObj);
-		tree(optObj);
-	} else {
-		//console.log("nothing to see here");
-	}
+		if (inputsGood()) {
+			let optObj = createOptionsObject();
+			//console.log("here's your options object", optObj);
+			tree(optObj);
+		}
+	} 
 }
 
 function buttonPressHandler() {
 	//console.log(event);
-	let optsObj = createOptionsObject();
-	//console.log("here's your options object", optsObj)
-	tree(optsObj);
+	if (inputsGood()) {
+		let optsObj = createOptionsObject();
+		//console.log("here's your options object", optsObj)
+		tree(optsObj);
+	} 
+}
+
+function inputsGood() {
+	if (charInput.value.length > 1) {
+		console.log("only use one character in the character field");
+	} else if (Number(heightInput.value) < 1 ) {
+		console.log("only integers greater than zero for the height field");
+	} else if (Number(heightInput.value) % 1 !== 0) {
+		console.log("no floating point numbers for the height field");
+	}	else {
+		return true;
+	}
 }
 
 function createOptionsObject() {
-	let optionsObj = {};
-	optionsObj.char = charInput.value;
-	optionsObj.height = Number(heightInput.value);
-	return optionsObj;
+		let optionsObj = {};	
```

Comment:
Be careful with whitespace. Inevitably it will lead to merge conflicts. You can change your settings in sublime to automatically trim the whitespace. Just enter this line into your custom settings:

`"trim_trailing_white_space_on_save": true,`

Also, a good way to visualize trailing whitespace is to add the Trailing Spaces package to your editor. It highlights any trailing whitespace in your code, making it super easy to stay on top of.